src/middlewared/middlewared/main.py

```diff
@@ -859,32 +862,41 @@ def get_schema(self, name):
             if asyncio.iscoroutinefunction(methodobj):
                 return await methodobj(*args)
             else:
-                return await self.threaded(methodobj, *args)
+                pool = None
+                if hasattr(methodobj, '_threaded'):
+                    pool = methodobj._threaded
+                if serviceobj._config.thread_pool:
+                    pool = serviceobj._config.thread_pool
```

Comment:
Maybe method-specific pool should have more priority over service-specific as it is more local?

Reply:
Agreed!